subsys/net/lib/lwm2m/lwm2m_rd_client.c

```diff
@@ -486,6 +486,13 @@ static int do_registration_reply_cb(const struct coap_packet *response,
 			client.server_ep);
 
 		return 0;
+	} else if (code == COAP_RESPONSE_CODE_CONTINUE) {
+#if defined(CONFIG_LWM2M_COAP_BLOCK_TRANSFER)
+		return 0;
+#else
+		LOG_ERR("Response code CONTINUE not supported");
+		ret = -EINVAL;
```

Comment:
This part seems not needed, `-EINVAL` is already set in `ret`.

Reply:
Also I wonder if this could be hidden somehow, so that the callback is only called when the callback is complete and the final response code is received.